src/MLOps.NET/Storage/Repositories/RunRepository.cs

```diff
@@ -169,5 +175,23 @@ public RegisteredModel GetLatestRegisteredModel(Guid experimentId)
 
             return this.registeredModelResolver.BuildEntity(db, latestRegisteredModel);
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="TInput"></typeparam>
+        /// <typeparam name="TOutput"></typeparam>
+        /// <param name="runId"></param>
+        public async Task RegisterSchema<TInput, TOutput>(Guid runId) where TInput : class where TOutput : class
+        {
+            using var db = this.contextFactory.CreateDbContext();
+            var modelInput = schemaGenerator.GenerateDefinition<TInput>("ModelInput");
+            var modelOutput = schemaGenerator.GenerateDefinition<TOutput>("ModelOutput");
+            var modelInputSchema = new ModelSchema(runId, typeof(TInput).Name, modelInput);
+            var modelOutputSchema = new ModelSchema(runId, typeof(TOutput).Name, modelOutput);
+            db.Schemas.Add(modelInputSchema);
```

Comment:
I think we should add a property to ModelSchema on a Run instance instead and in this method just get the run and update it by setting this property